In [6]:
import json 
import numpy as np
from collections import Counter

with open('retrieval_base_layer7_bert-base-uncased_layer0_p4_k100.json', 'r') as f:
    encodecmae_bert = json.load(f)

with open('retrieval_wav2vec2_layer6_bert-base-uncased_layer0_p4_k100.json', 'r') as f:
    w2v_bert = json.load(f)


# Palabras bien devueltas

In [3]:
keys_test = [encodecmae_bert['all_values'][i] for i in encodecmae_bert['rows_deleted']]
total = len(keys_test)
for i in range(total):
    word = keys_test[i]
    k_1 = encodecmae_bert['retrieval_indices'][i][0]
    if word == encodecmae_bert['keys_anchors'][k_1]:
        print(f"word: {word} retrieval:{encodecmae_bert['keys_anchors'][k_1]}")    
        print('---------')


word: some retrieval:some
---------
word: of retrieval:of
---------
word: we retrieval:we
---------
word: was retrieval:was
---------
word: me retrieval:me
---------
word: in retrieval:in
---------
word: river retrieval:river
---------
word: so retrieval:so
---------
word: keep retrieval:keep
---------
word: spirit retrieval:spirit
---------
word: of retrieval:of
---------
word: the retrieval:the
---------
word: we retrieval:we
---------
word: that retrieval:that
---------
word: have retrieval:have
---------
word: do retrieval:do
---------
word: that retrieval:that
---------
word: some retrieval:some
---------
word: not retrieval:not
---------
word: the retrieval:the
---------
word: these retrieval:these
---------
word: things retrieval:things
---------
word: give retrieval:give
---------
word: the retrieval:the
---------
word: food retrieval:food
---------
word: in retrieval:in
---------
word: how retrieval:how
---------
word: could retrieval:could
---------
word: were retrieval:were


In [4]:
keys_test = [w2v_bert['all_values'][i] for i in w2v_bert['rows_deleted']]
total = len(keys_test)
for i in range(total):
    word = keys_test[i]
    k_1 = w2v_bert['retrieval_indices'][i][0]
    if word == w2v_bert['keys_anchors'][k_1]:
        print(f"word: {word} retrieval:{w2v_bert['keys_anchors'][k_1]}")    
        print('---------')


word: some retrieval:some
---------
word: this retrieval:this
---------
word: of retrieval:of
---------
word: fire retrieval:fire
---------
word: we retrieval:we
---------
word: never retrieval:never
---------
word: illustration retrieval:illustration
---------
word: was retrieval:was
---------
word: me retrieval:me
---------
word: probably retrieval:probably
---------
word: in retrieval:in
---------
word: garden retrieval:garden
---------
word: pocket retrieval:pocket
---------
word: voice retrieval:voice
---------
word: an retrieval:an
---------
word: keep retrieval:keep
---------
word: that retrieval:that
---------
word: river retrieval:river
---------
word: for retrieval:for
---------
word: during retrieval:during
---------
word: so retrieval:so
---------
word: their retrieval:their
---------
word: keep retrieval:keep
---------
word: country retrieval:country
---------
word: like retrieval:like
---------
word: tore retrieval:tore
---------
word: office retrieval:office
---------
wo

# Palabras mal devueltas

Consideramos palabras mal devueltas si no aparece la palabra que deberia en el top 5

In [4]:
keys_test = [encodecmae_bert['all_values'][i] for i in encodecmae_bert['rows_deleted']]
total = len(keys_test)
words_encodec = []
for i in range(total):
    word = keys_test[i]
    k_1 = encodecmae_bert['retrieval_indices'][i]
    if not any(encodecmae_bert['keys_anchors'][index] == word for index in k_1 if index < len(encodecmae_bert['keys_anchors'])):
        word_retrieval =  [encodecmae_bert['keys_anchors'][k_1] for k_1 in encodecmae_bert['retrieval_indices'][i]]
        words_encodec.append([word, word_retrieval])
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')

word: this retrieval:['the', 'the', 'the', 'the', 'the'], indice 1
---------
word: fire retrieval:['are', 'are', 'are', 'are', 'are'], indice 3
---------
word: never retrieval:['am', 'am', 'am', 'am', 'am'], indice 5
---------
word: excitement retrieval:['nile', 'wonder', 'general', 'autumn', 'em'], indice 9
---------
word: probably retrieval:['wildly', 'little', 'little', 'little', 'little'], indice 10
---------
word: cross retrieval:['crowd', 'heart', 'hot', 'karl', 'karl'], indice 12
---------
word: and retrieval:['in', 'in', 'in', 'in', 'in'], indice 13
---------
word: no retrieval:['mode', 'mode', 'mode', 'mode', 'mode'], indice 15
---------
word: pocket retrieval:['back', 'up', 'up', 'up', 'back'], indice 16
---------
word: voice retrieval:['oil', 'war', 'ourselves', 'war', 'war'], indice 17
---------
word: an retrieval:['in', 'in', 'in', 'in', 'in'], indice 18
---------
word: keep retrieval:['he', 'he', 'he', 'he', 'he'], indice 19
---------
word: recipe retrieval:['be', 'be', '

In [5]:
keys_test = [w2v_bert['all_values'][i] for i in w2v_bert['rows_deleted']]
total = len(keys_test)
words_w2v = []
for i in range(total):
    word = keys_test[i]
    k_1 = w2v_bert['retrieval_indices'][i]
    if not any(w2v_bert['keys_anchors'][index] == word for index in k_1 if index < len(w2v_bert['keys_anchors'])):
        word_retrieval =  [w2v_bert['keys_anchors'][k_1] for k_1 in w2v_bert['retrieval_indices'][i]]
        words_w2v.append([word, word_retrieval])
        print(f"word: {word} retrieval:{word_retrieval}, indice {i}")    
        print('---------')

word: cross retrieval:['across', 'across', 'across', 'across', 'across'], indice 12
---------
word: no retrieval:['now', 'now', 'now', 'now', 'now'], indice 15
---------
word: recipe retrieval:['he', 'he', 'he', 'he', 'he'], indice 20
---------
word: wielded retrieval:['sealed', 'little', 'little', 'little', 'little'], indice 34
---------
word: drouet retrieval:['herbert', 'ardour', 'ardor', 'herbert', 'bread'], indice 37
---------
word: panada retrieval:['at', 'at', 'canada', 'i', 'add'], indice 44
---------
word: quarters retrieval:['corner', 'quarter', 'quarter', 'corner', 'court'], indice 50
---------
word: olden retrieval:['old', 'old', 'old', 'old', 'old'], indice 54
---------
word: call retrieval:['called', 'called', 'called', 'called', 'called'], indice 55
---------
word: paid retrieval:['pay', 'pay', 'pay', 'pay', 'pay'], indice 57
---------
word: venus retrieval:['in', 'in', 'in', 'in', 'in'], indice 61
---------
word: while retrieval:['wild', 'wild', 'wild', 'wild', 'wild'],

¿Cuántas palabras hay "mal devueltas" en cada uno?

In [11]:
len(keys_test)

7035

In [9]:
len(words_w2v)

1812

In [10]:
len(words_encodec)

4500

In [22]:
words_w2v_counter = [item[0] for item in words_w2v]
words_encodec_counter = [item[0] for item in words_encodec]

# Count each word in both lists
counter_w2v = Counter(words_w2v_counter)
counter_encodec = Counter(words_encodec_counter)

# Find the intersection of words in both lists
common_words = set(words_w2v_counter) & set(words_encodec_counter)

# Print the results
print("Words in words_w2v:", len(counter_w2v))
print("Words in words_encodec:", len(counter_encodec))
print("Common words:", common_words)
print("Number of common words:", len(common_words))

Words in words_w2v: 1532
Words in words_encodec: 2316
Common words: {'caisson', 'wall', 'hands', 'exchanged', 'rejects', 'hello', 'cooking', 'knowing', 'commercial', 'thrust', 'preparing', 'catching', 'protruded', 'services', 'sight', 'carries', 'does', 'cover', 'beyond', 'operation', 'quixote', 'raising', 'hypocrites', 'dares', 'moderate', 'sending', 'luxuriously', 'pleasing', 'multitude', 'folks', 'stored', 'specialty', 'fast', 'paid', 'soil', 'crying', 'savour', 'apartment', 'readers', 'alma', 'exact', 'fee', 'depended', 'mark', 'stern', 'leaves', 'jeering', 'unhappy', 'implicitly', 'why', 'attached', 'sprawling', 'unlike', 'polychrome', 'transition', 'aristocratic', 'wagons', 'border', 'containing', 'complacently', 'consider', 'everywhere', 'startled', 'position', 'acquaintance', 'entirety', 'unfastening', 'horseback', 'lesson', 'deftly', 'accepted', 'bitterness', 'mouths', 'difficult', 'pleased', 'badgers', 'continues', 'log', 'foul', 'meadows', 'hearts', 'occasionally', 'watcher'

Veamos las palabras que habia que devolver cuántas apariciones tiene

In [23]:
with open('../../words_counts.json', 'r') as json_file:
    words_count = json.load(json_file)

In [24]:
common_word_counts = {word: words_count.get(word, 0) for word in keys_test}
count_one_value = sum(1 for count in common_word_counts.values() if count == 1)

print("Number of words with a count of 1 on key_test:", count_one_value)

Number of words with a count of 1 on key_test: 430


In [25]:
common_word_counts = {word: words_count.get(word, 0) for word in common_words}
count_one_value = sum(1 for count in common_word_counts.values() if count == 1)

print("Number of words with a count of 1 on common_words:", count_one_value)

Number of words with a count of 1 on common_words: 430


Hay 430 palabras que no habia visto el encoder ¿Qué devuelve para esas palabras?

In [38]:
count = 0
words_encodec_unique = []
for i in range(len(words_encodec)):
    word = words_encodec[i][0]
    if words_count[word] == 1:
        words_encodec_unique.append(words_encodec[i])
        count += 1
        print(f"word: {word} retrieval:{words_encodec[i][1]}, indice {i}")    

word: wielded retrieval:['little', 'little', 'little', 'little', 'little'], indice 22
word: performance retrieval:['from', 'from', 'from', 'from', 'from'], indice 43
word: charwoman retrieval:['home', 'matter', 'hour', 'home', 'wrong'], indice 48
word: darker retrieval:['are', 'are', 'are', 'are', 'are'], indice 62
word: appointment retrieval:['room', 'me', 'room', 'me', 'suppressed'], indice 65
word: woven retrieval:['warm', 'more', 'warm', 'ball', 'more'], indice 68
word: bustle retrieval:['up', 'up', 'up', 'up', 'up'], indice 74
word: distinctively retrieval:['to', 'ceiling', 'to', 'to', 'to'], indice 89
word: ampler retrieval:['them', 'them', 'them', 'them', 'them'], indice 91
word: pales retrieval:['hoe', 'rows', 'have', 'have', 'have'], indice 102
word: ancestral retrieval:['on', 'on', 'on', 'on', 'on'], indice 103
word: administrator retrieval:['in', 'in', 'in', 'in', 'in'], indice 112
word: lantern retrieval:['room', 'him', 'him', 'him', 'him'], indice 115
word: usurping retrie

In [27]:
print(count)

430


In [39]:
count = 0
words_w2v_unique = []
for i in range(len(words_w2v)):
    word = words_w2v[i][0]
    if words_count[word] == 1:
        words_w2v_unique.append(words_w2v[i])
        count += 1
        print(f"word: {word} retrieval:{words_w2v[i][1]}, indice {i}")    

word: wielded retrieval:['sealed', 'little', 'little', 'little', 'little'], indice 3
word: performance retrieval:['reform', 'for', 'for', 'for', 'for'], indice 13
word: charwoman retrieval:['bun', 'door', 'room', 'them', 'them'], indice 14
word: darker retrieval:['dark', 'dark', 'dark', 'dark', 'dark'], indice 18
word: appointment retrieval:['drainage', 'marriage', 'complained', 'moment', 'meantime'], indice 19
word: woven retrieval:['womb', 'rudely', 'wave', 'wave', 'wave'], indice 20
word: bustle retrieval:['bells', 'buffalo', 'robust', 'solve', 'mall'], indice 21
word: distinctively retrieval:['indistinctly', 'existing', 'indistinct', 'be', 'be'], indice 25
word: ampler retrieval:['man', 'man', 'man', 'man', 'man'], indice 26
word: pales retrieval:['of', 'of', 'of', 'of', 'of'], indice 29
word: ancestral retrieval:['senses', 'external', 'thence', 'mosaic', 'remonstrances'], indice 30
word: administrator retrieval:['witted', 'in', 'in', 'in', 'in'], indice 35
word: lantern retrieval:

In [42]:
words_w2v_counter = [item[0] for item in words_w2v_unique]
words_encodec_counter = [item[0] for item in words_encodec_unique]

# Find the intersection of words in both lists
common_words = set(words_w2v_counter) & set(words_encodec_counter)

# Creating DataFrame with word lists
data = {
    'word': list(common_words),
    'encodec': [next(item[1] for item in words_encodec_unique if item[0] == word) for word in common_words],
    'w2v': [next(item[1] for item in words_w2v_unique if item[0] == word) for word in common_words]
}

df_unique = pd.DataFrame(data)

df_unique


word                            encodec  \
0    temperate          [her, her, her, her, her]   
1      caisson          [say, say, say, say, say]   
2         tait          [day, day, day, day, day]   
3        wench          [way, way, way, way, way]   
4     disorder             [or, were, or, or, or]   
..         ...                                ...   
425  patriarch          [are, are, are, are, are]   
426       curl          [her, her, her, her, her]   
427  dalliance          [man, man, man, man, man]   
428    cursing  [first, first, first, first, are]   
429   civilian              [in, in, in, the, in]   

                                      w2v  
0              [lamp, her, her, her, her]  
1                    [on, on, on, on, on]  
2               [day, day, day, day, day]  
3       [leash, parish, me, hand, teepee]  
4     [beer, order, door, fear, reviewer]  
..                                    ...  
425  [hair, tartaric, hair, gifted, hair]  
426       [occur, girl, girl, girl, girl]  
427             [lying, on, mell, on, on]  
428  [curses, sutter, sure, first, first]  
429             [the, the, the, the, the]  

[430 rows x 3 columns]

¿Qué pasa con las demás palabras?

In [31]:
words_encodec_not_unique = []
for i in range(len(words_encodec)):
    word = words_encodec[i][0]
    k_1 = w2v_bert['retrieval_indices'][i]
    if not words_count[word] == 1:
        words_encodec_not_unique.append(words_encodec[i])
        print(f'word: {word}, retrieval: { words_encodec[i][1]}')

word: this, retrieval: ['the', 'the', 'the', 'the', 'the']
word: fire, retrieval: ['are', 'are', 'are', 'are', 'are']
word: never, retrieval: ['am', 'am', 'am', 'am', 'am']
word: excitement, retrieval: ['nile', 'wonder', 'general', 'autumn', 'em']
word: probably, retrieval: ['wildly', 'little', 'little', 'little', 'little']
word: cross, retrieval: ['crowd', 'heart', 'hot', 'karl', 'karl']
word: and, retrieval: ['in', 'in', 'in', 'in', 'in']
word: no, retrieval: ['mode', 'mode', 'mode', 'mode', 'mode']
word: pocket, retrieval: ['back', 'up', 'up', 'up', 'back']
word: voice, retrieval: ['oil', 'war', 'ourselves', 'war', 'war']
word: an, retrieval: ['in', 'in', 'in', 'in', 'in']
word: keep, retrieval: ['he', 'he', 'he', 'he', 'he']
word: recipe, retrieval: ['be', 'be', 'be', 'be', 'be']
word: that, retrieval: ['the', 'the', 'the', 'the', 'the']
word: for, retrieval: ['were', 'were', 'were', 'were', 'were']
word: during, retrieval: ['are', 'are', 'are', 'are', 'are']
word: their, retrieval

In [34]:
words_w2v_not_unique = []
for i in range(len(words_w2v)):
    word = words_w2v[i][0]
    if not words_count[word] == 1:
        words_w2v_not_unique.append(words_w2v[i])
        print(f'word: {word}, retrieval: { words_w2v[i][1]}')

word: cross, retrieval: ['across', 'across', 'across', 'across', 'across']
word: no, retrieval: ['now', 'now', 'now', 'now', 'now']
word: recipe, retrieval: ['he', 'he', 'he', 'he', 'he']
word: drouet, retrieval: ['herbert', 'ardour', 'ardor', 'herbert', 'bread']
word: panada, retrieval: ['at', 'at', 'canada', 'i', 'add']
word: quarters, retrieval: ['corner', 'quarter', 'quarter', 'corner', 'court']
word: olden, retrieval: ['old', 'old', 'old', 'old', 'old']
word: call, retrieval: ['called', 'called', 'called', 'called', 'called']
word: paid, retrieval: ['pay', 'pay', 'pay', 'pay', 'pay']
word: venus, retrieval: ['in', 'in', 'in', 'in', 'in']
word: while, retrieval: ['wild', 'wild', 'wild', 'wild', 'wild']
word: edition, retrieval: ['tuition', 'gifted', 'to', 'to', 'to']
word: nor, retrieval: ['or', 'or', 'or', 'or', 'or']
word: beaten, retrieval: ['beat', 'beat', 'beat', 'beat', 'beat']
word: whom, retrieval: ['who', 'who', 'who', 'who', 'who']
word: its, retrieval: ['the', 'the', 'th

In [43]:
import pandas as pd
words_w2v_counter = [item[0] for item in words_w2v_not_unique]
words_encodec_counter = [item[0] for item in words_encodec_not_unique]

# Find the intersection of words in both lists
common_words = set(words_w2v_counter) & set(words_encodec_counter)

# Creating DataFrame with word lists
data = {
    'word': list(common_words),
    'encodec': [next(item[1] for item in words_encodec_not_unique if item[0] == word) for word in common_words],
    'w2v': [next(item[1] for item in words_w2v_not_unique if item[0] == word) for word in common_words]
}

df_not_unique = pd.DataFrame(data)

df_not_unique

word                                        encodec  \
0       circuit                 [sort, sort, sort, sort, said]   
1          wall                      [all, all, all, all, all]   
2     oppressed  [coughed, thrilled, best, simmonds, thoughts]   
3       possess                           [is, is, is, is, is]   
4        county                      [her, her, her, her, her]   
...         ...                                            ...   
1086     simmer                     [are, are, are, are, were]   
1087   stations                      [she, she, she, she, she]   
1088        has                      [his, his, his, his, his]   
1089       free             [freed, stacy, stacy, stacy, rear]   
1090       aunt                           [on, on, on, on, on]   

                                                w2v  
0                         [her, her, her, her, her]  
1                         [all, all, all, all, all]  
2       [pressed, pressed, pressed, crept, pressed]  
3                        [assessed, as, as, as, as]  
4     [country, country, country, country, country]  
...                                             ...  
1086         [sense, missus, sense, missus, missus]  
1087             [stayed, seen, stated, these, see]  
1088                           [is, is, is, is, is]  
1089             [afraid, few, afraid, few, afraid]  
1090                           [on, on, on, on, on]  

[1091 rows x 3 columns]